pkg/core/image_mapper.go

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2018 The original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package core
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+)
+
+const (
+	dockerHubHost     = "docker.io"
+	fullDockerHubHost = "index.docker.io"
+)
+
+// imageMapper does substring replacement of images with minimal delimiter checking.
+// It does not understand the full (YAML) syntax of the input string.
+// It certainly does not understand the semantics of the input string.
+type imageMapper struct {
+	mappings map[string]string
+}
+
+func newImageMapper(mappedHost string, mappedUser string, images []string) (*imageMapper, error) {
+	if err := contains(mappedHost, "/", `"`, " "); err != nil {
```

Comment:
Not clear immediately what this does. Maybe rename to `containsAny`?

Reply:
Wouldn't it be better to actually validate this as a dns entry altogether? We have the technology, we have the capability to make the world's first idiot proof cli...